packages/flutter/test/painting/text_span_test.dart

```diff
@@ -476,4 +477,156 @@ void main() {
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 1000))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 0, end: 0))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 0, end: 1))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 1000)), (1, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 1, end: 2))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (1, TextStyle(fontSize: 1000)), (2, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 2, end: 5))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (2, TextStyle(fontSize: 1000)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 0, end: 5))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 1000)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 5, end: 6))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 1000)), (6, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 6, end: 6))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 6, end: 7))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10)), (6, TextStyle(fontSize: 1000)), (7, TextStyle(fontSize: 10))],
+      );
+      expect(
+        toStyleRuns(simpleSpan.updateAttributes(const InlineSpanAttributes(fontSize: 1000), const TextRange(start: 6, end: 10))),
+        const <(int, TextStyle)>[(0, TextStyle(fontSize: 100)), (5, TextStyle(fontSize: 10)), (6, TextStyle(fontSize: 1000))],
+      );
+    });
+
+    test('can add and delete recognizers', () {
+      final GestureRecognizer recognizer = TapGestureRecognizer();
+      addTearDown(recognizer.dispose);
+      final int length = simpleSpan.toPlainText().length;
+      final TextSpan withRecognizer = simpleSpan.updateAttributes(
+        InlineSpanAttributes(recognizer: Left(recognizer)),
```

Comment:
As stated elsewhere, the `Left` in there feels out of place. When reading this code, it is unclear what this means and if you click through on `Left` it gives you a very generic description about Left that isn't helpful. In fact, I don't even know what docs to point people to if they want to understand what this Left instantiation here does.